pkg/kubelet/kubelet_pods.go

```diff
@@ -1458,6 +1458,14 @@ func getPhase(spec *v1.PodSpec, info []v1.ContainerStatus) v1.PodPhase {
 // internal pod status. This method should only be called from within sync*Pod methods.
 func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.PodStatus) v1.PodStatus {
 	klog.V(3).InfoS("Generating pod status", "pod", klog.KObj(pod))
+
+	// ensure the probe managers have up-to-date status for containers
+	initContainerIDs := sets.New[string]() // FIXME find a better way to recognize an initContainer
+	for _, is := range pod.Status.InitContainerStatuses {
+		initContainerIDs.Insert(is.ContainerID)
+	}
```

Comment:
This looks wrong. `status.InitContainerStatuses` is not initialized yet.

With this code, initContainerIDs is always empty.

Reply:
Yeah, I thought about that too... however it fails if you have a name collision between `containers` and `initContainers`.

Reply:
The names of all containers(including initContainers) are unique as far as I know.

Reply:
Indeed, you're right ! Updating my PR, thanks for the solution !